common/OpTestUtil.py

```diff
@@ -2571,6 +2571,12 @@ def login(self, username=None, password=None):
                 self.xAuthHeader['X-Auth-Token'] = match.group(1)
                 self.jsonHeader.update(self.xAuthHeader)
             json_data = json.loads(r.text)
+            if not (json_data.get('token') is None):
```

Comment:
Can you fix the commit message format? Also the message header has a typo (EMB). I guess you meant eBMC.

Instead of hardcoding the expected response can we enhance the check to include token as a valid successful response? Will that work?

if (json_data['status'] != "ok" or json_data.get('token') is None ):

Reply:
The above check should use `and` instead of `or`

Reply:
thank @sacsant for review, will respin with valid commit ,
Sachin can be done with some tweak (as json_data.get('token') is None will not work ) but i feel current check is more than enough as older firmware we did not get Token, so idea like if Token key is not there in repose body (not None) go and have populate JSON data to have integrity of flow

Reply:
fixed commit message @sacsant please have look , and can you please my response for other part of review comment and let me know if you are ok with this

Reply:
> and this will error out (json_data['status']  as key not present 
ah yes.

Reply:
> fixed commit message @sacsant please have look , and can you please check my response for other part of review comment and let me know if you are ok with this

How about
OpTestUtils: Handle successful login response format difference

Depending on code level running on BMC/eBMC, expected format
for a response on successful login operation may differ. 

Expected: `status': 'ok'`
Actual: `"token":"<token>"`

This patch adds code to handle this valid response.

Reply:
@sacsant code handle that already as, so I don't feel we need to unnecessarily introduce complexity. As in case of successful login only response have token only get other case it go with op-test usually fail flow 

```
common.Exceptions.HTTPCheck: Something happened with the HTTP Server.  Review the following for more details
Message="Requests post problem, check that your credentials are properly setup URL=https://eBMC/login username=admin password=s0penBmc!, Exception=Something happened with the HTTP Server.  Review the following for more details
```

Reply:
Sorry, I wasn't clear in my previous update. The text in previous update is actually a suggestion for commit message. I am not asking you to change the logic, I am good with your proposed logic.

Reply:
Ah , thanks for suggestion , taken care , please have look